script/node_bs.py

```diff
@@ -0,0 +1,281 @@
+#ROOT = os.path.dirname(os.path.dirname(__file__))
+ROOT = os.getcwd()
+package_name = input("Enter Package name (Package name should match with the directory name): ")
+#package_name = 'elasticsearch'
+package_name = package_name.lower()
+dir_name = f"{ROOT}{path_separator}{package_name[0]}{path_separator}{package_name}"
+
+
+github_url=''
+latest_release=''
+active_repo=False
+new_build_script=''
+branch_pkg=""
+
+
+def get_latest_build_script(dir_name):
+    files_path = os.path.join(dir_name,"*")
+    files = sorted(glob.iglob(files_path),key=os.path.getctime,reverse=True)
+    latest_existing_build_script =""
+    for script_file in files:
+        if script_file.endswith(".sh") and (script_file.find("ubuntu")==-1):
+            global github_url
+            github_url = get_repo_url(script_file)
+            global active_repo
+            if github_url!="NUrl" and github_url!='':
+                if (check_repo_activeness(github_url)):
+                    active_repo = True
+                    global latest_release
+                    latest_release = get_latest_release(github_url)
+                else:
+                    active_repo = False
+                    print("\n Repo is Not active . Stopping the further process.")
+                    exit()
+
+            return script_file
+
+    return "NPresent"
+
+def get_repo_url(script_file):
+    with open(script_file,'r',encoding='utf-8') as f:
+        contents=f.readlines()
+        for line in contents:
+            if line.startswith('# Source repo') :
+                github_url = ":".join(line.split(':')[1:]).strip()
+                return github_url
+    return "NUrl"
+
+def check_repo_activeness(package_url):
+    owner, repo = package_url.replace('.git','').split('/')[-2:]
+    active_response=requests.get(GITHUB_PACKAGE_INFO_API.format(owner,repo,'branches','master'))
+
+   
+    last_commit_date=active_response.json()["commit"]["commit"]["committer"]["date"]
+    last_commit_year=last_commit_date.split('-')[0]
+    today_date=str(date.today())
+    today_date=today_date.split('-')[0]
+
+    if (int(today_date) - int(last_commit_year))>=3:
+        #print(f"{package_url} Package Not Active")
+        return False
+    return True
+
+def get_latest_release(package_url):
+    owner, repo = package_url.replace('.git','').split('/')[-2:]
+    #response = requests.get(GITHUB_PACKAGE_INFO_API.format(owner, repo)).json()
+    #release_github_url=GITHUB_PACKAGE_INFO_API+'/latest/releases'
+    print("\n Owner :",owner)
+    print("\n Repo  :",repo)
+    response = requests.get(GITHUB_PACKAGE_INFO_API.format(owner,repo,'releases','latest'))
+    if response.status_code!=200:
+        print("\n Release Not Present")
+        #print("type",type(response.status_code))
+    
+        #print("Printing Response",response)
+    
+        response_tag=requests.get("https://api.github.com/repos/{}/{}/{}/{}/{}".format(owner,repo,'git','refs','tags'))
+        #print(response_tag)
```

Comment:
This API returns the subset of the tags, in alphabetical order, so there is a possibility that the tag found might not be the latest tag.